plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java

```diff
@@ -240,55 +253,151 @@ public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandl
         mongoSession.dropColumn(((MongoTableHandle) tableHandle).getSchemaTableName(), ((MongoColumnHandle) column).getName());
     }
 
+    private String generateTemporaryTableName()
+    {
+        return "tmp_trino_" + UUID.randomUUID().toString().replace("-", "");
+    }
+
     @Override
     public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorTableLayout> layout, RetryMode retryMode)
     {
-        if (retryMode != NO_RETRIES) {
-            throw new TrinoException(NOT_SUPPORTED, "This connector does not support query retries");
-        }
-
         List<MongoColumnHandle> columns = buildColumnHandles(tableMetadata);
-
         mongoSession.createTable(tableMetadata.getTable(), columns, tableMetadata.getComment());
+        List<MongoColumnHandle> handleColumns = columns.stream().filter(c -> !c.isHidden()).collect(toImmutableList());
```

Comment:
Rename `c` to `column`. Please avoid abbreviations as much as possible. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-abbreviations